### ISSUE TITLE
check 'PQgetResult(this->connection) == NULL' failed

### ISSUE THREAD

**Issue by dtseiler** [pgbackrest/pgbackrest#1233]
Please provide the following information when submitting an issue (feature requests or general comments can skip this):

1. pgBackRest version: 2.28

2. PostgreSQL version: 10.15

3. Operating system/version - if you have more than one server (for example, a database server, a repository host server, one or more standbys), please specify each: Ubunto 18.04 (MS Azure image)

4. Did you install pgBackRest from source or from a package? From PGDG apt package

5. Please attach the following as applicable:
    - `pgbackrest.conf` file(s)
    - `postgresql.conf` settings applicable to pgBackRest (`archive_command`, `archive_mode`, `listen_addresses`, `max_wal_senders`, `wal_level`, `port`)
    - errors in the postgresql log file before or during the time you experienced the issue
    - log file in `/var/log/pgbackrest` for the commands run (e.g. `/var/log/pgbackrest/mystanza_backup.log`)

7. Describe the issue:

Had an incremental backup fail tonight while waiting for the initial checkpoint:

```
2020-11-18 01:00:02.058 P00   INFO: backup command begin 2.28: --compress-type=lz4 --log-level-console=info --log-level-file=info --pg1-path=/var/lib/postgresql/10/main --pg1-port=5432 --process-max=4 --repo1-azure-account=<redacted> --repo1-azure-container=foo --repo1-azure-key=<redacted> --repo1-cipher-pass=<redacted> --repo1-cipher-type=aes-256-cbc --repo1-path=/foo --repo1-retention-full=2 --repo1-type=azure --stanza=postgres-foo --type=incr
2020-11-18 01:00:03.394 P00   INFO: last backup label = 20201115-010001F_20201117-010002I, version = 2.28
2020-11-18 01:00:03.394 P00   INFO: execute non-exclusive pg_start_backup(): backup begins after the next regular checkpoint completes
2020-11-18 01:00:41.132 P00 ASSERT: [025]: check 'PQgetResult(this->connection) == NULL' failed
                                    STACK TRACE:
                                    postgres/client:pgClientQuery:326:(debug log level required for parameters)
                                    db/db:dbQuery:(debug log level required for parameters)
                                    db/db:dbQueryRow:(debug log level required for parameters)
                                    db/db:dbBackupStart:(debug log level required for parameters)
                                    command/backup/backup:backupStart:(debug log level required for parameters)
                                    command/backup/backup:cmdBackup:(debug log level required for parameters)
                                    main:main:(debug log level required for parameters)
2020-11-18 01:00:41.132 P00   INFO: http statistics: objects 1, sessions 1, requests 13, retries 0, closes 0
2020-11-18 01:00:41.132 P00   INFO: backup command end: aborted with exception [025]
```
I've never seen this error before. We've been running these same backups to Azure for a couple months now (weekly full, nightly incr).

I've kicked off another incremental manually that seems to be going. Might be a hiccup but would like to know if there's something to tune here. We were slamming this particular database with sessions earlier in the day as part of a migration project.

**Comment by dtseiler** [comment]
This is probably on our end. It looks like some automatic process ran the patch upgrade from PG 10.14 to 10.15, and the apt packages automatically restart the DB when you do that. We made a point of excluding PGDG packages from unattended upgrades so I'm not sure how this got through, and why it's only affecting this one out of our 22 clusters. I'll check back in the morning.

**Comment by dwsteele** [comment]
Yeah, this was caused by the server dropping the connection, but the logic does not seem quite right here. I think we need a loop here to clear any remaining results to avoid an assert. Or maybe even check each result for an error.

You would have gotten an error anyway, but it should not have been an assert.

**Comment by dtseiler** [comment]
So I learned that this month we enabled unattended upgrades on these Ubuntu 18.04 VMs. By default these exclude PGDG packages. However Ubuntu 18.04 does provide `postgresql-10` packages as well and when the upgrade ran it saw the the Ubuntu package at 10.15 while the PGDG package was at 10.14 and so it upgraded. Apparently excluding PGDG doesn't prevent this (but would prevent it from upgrading *to* PGDG, I'd wager. It's only a problem with postgresql-10 on Ubuntu 18.04.

Anyway, closing this. Thanks.

**Comment by dwsteele** [comment]
Leaving this open so we can improve the error message.

**Comment by dwsteele** [comment]
The error message was improved in 23e966cd8f1.
